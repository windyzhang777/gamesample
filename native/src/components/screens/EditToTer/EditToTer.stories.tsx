import React from 'react';
import {storiesOf} from '@storybook/react';
import EditToTer from '.';
import Provider from '../../../utils/Provider';
import {useSelector} from 'react-redux';

storiesOf('Edit ToTer', module)
  .addDecorator((story) => <Provider story={story()} />)
  .add('Normal Edit ToTer', () => {
    return (
      <EditToTer
        useSelector={useSelector}
        match={{
          params: {
            toterId: 'toter_1',
          },
        }}
      />
    );
  });
