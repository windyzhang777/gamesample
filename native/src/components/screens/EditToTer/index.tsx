/* eslint-disable react-hooks/rules-of-hooks */
import React from 'react';
import EditToTerShared, {
  EditToTerProps,
  EditToTerRenderProps,
} from 'shared/containers/screens/EditToTer';
import {View, Text, Button} from 'react-native';
import {useSelector} from 'react-redux';

export default function EditToTer(props: EditToTerProps) {
  return (
    <EditToTerShared
      {...props}
      useSelector={useSelector}
      renderComponent={({toter, goBackCallback}: EditToTerRenderProps) => {
        return (
          <View>
            <Button title={'<'} onPress={goBackCallback} />
            <Text>ToTerName: {toter.toterName}</Text>
            <Text>ToTerId: {toter.toterId}</Text>
          </View>
        );
      }}
    />
  );
}
