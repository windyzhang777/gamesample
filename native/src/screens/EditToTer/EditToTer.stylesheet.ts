import {StyleSheet} from 'react-native';

export default StyleSheet.create({
  EditToTer: {
    backgroundColor: '#5c446f',
    height: '100%',
  },
  text: {
    color: 'white',
  },
});
