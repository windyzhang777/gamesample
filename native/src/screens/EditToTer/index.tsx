import React from 'react';
import {View, Text} from 'react-native';
import styles from './EditToTer.stylesheet';
import {ToTerType} from 'src/store/trick-or-treat/types';
import {useSelector} from 'react-redux';

export default function EditToTer({route}) {
  const toter = useSelector((state: any) =>
    state.toterList.toterItems.find((toter: ToTerType) => toter.toterId === route.params.toterId),
  );

  return (
    <View style={styles.EditToTer}>
      <Text style={styles.text}>ToTerName: {toter.toterName}</Text>
      <Text style={styles.text}>ToTerId: {toter.toterId}</Text>
    </View>
  );
}
