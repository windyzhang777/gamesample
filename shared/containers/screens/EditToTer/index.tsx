import {RenderedComponent} from 'shared/models/RenderedComponent';
import {ToTerType} from '../../../store/trick-or-treat/types';

export interface EditToTerProps {
  useSelector: any;
  history?: any;
  match: any;
}

export interface EditToTerRenderProps {
  toter: ToTerType;
  goBackCallback: () => void;
}

export default function EditToTer({
  useSelector,
  match,
  history,
  renderComponent,
}: EditToTerProps & RenderedComponent<EditToTerRenderProps>) {
  //might want to key these and just do a hash lookup. O(N)
  const toter = useSelector((state: any) =>
    state.toterList.toterItems.find((toter: ToTerType) => toter.toterId === match.params.toterId),
  );
  const goBackCallback = () => {
    if (history !== undefined) {
      history.goBack();
    } else {
      console.warn('History is not present on the editrorer, can not resolve go back path');
    }
  };
  const renderProps: EditToTerRenderProps = {
    goBackCallback,
    toter: toter,
  };
  return renderComponent(renderProps);
}
