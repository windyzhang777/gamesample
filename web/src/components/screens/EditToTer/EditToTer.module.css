.EditToTer {
  background-color: #5c446f;
  height: 100%;
  color: #fff;
  font-size: 18px;
  font-weight: bold;
  text-align: center;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  box-sizing: border-box;
}
