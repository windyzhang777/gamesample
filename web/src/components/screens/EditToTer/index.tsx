import React from 'react';
import EditToTerShared, {
  EditToTerProps,
  EditToTerRenderProps,
} from 'shared/containers/screens/EditToTer';
import classes from './EditToTer.module.css';
import {useSelector} from 'react-redux';

export default function EditToTer(props: EditToTerProps) {
  return (
    <EditToTerShared
      {...props}
      useSelector={useSelector}
      renderComponent={({toter, goBackCallback}: EditToTerRenderProps) => {
        return (
          <div className={classes.EditToTer}>
            <button onClick={goBackCallback}>{'<'}</button>
            <div>{toter.toterName}</div>
          </div>
        );
      }}
    />
  );
}
