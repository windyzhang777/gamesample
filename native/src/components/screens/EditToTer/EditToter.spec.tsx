import React from 'react';
import {shallow} from 'enzyme';
import EditToTer from './index';

describe('Home Component', () => {
  it('renders wrapper', () => {
    expect(shallow(<EditToTer />)).toMatchSnapshot();
  });
  it('renders component with a display name', () => {
    const RenderPropComponent = shallow(<EditToTer />).prop('renderComponent');
    expect(
      shallow(
        <RenderPropComponent
          toter={{toterName: 'Trick or Treater Name', toterId: 'ToTer ID'}}
          goBackCallback="function generated from to be called on back"
        />,
      ),
    ).toMatchSnapshot();
  });
});
