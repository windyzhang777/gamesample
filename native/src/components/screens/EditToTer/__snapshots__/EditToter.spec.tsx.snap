// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Component renders component with a display name 1`] = `
<View>
  <Button
    onPress="function generated from to be called on back"
    title="<"
  />
  <Text>
    ToTerName: 
    Trick or Treater Name
  </Text>
  <Text>
    ToTerId: 
    ToTer ID
  </Text>
</View>
`;

exports[`Home Component renders wrapper 1`] = `
<EditToTer
  renderComponent={[Function]}
  useSelector={[Function]}
/>
`;
